### PR TITLE
Remove Parent tests/impls/reqs

### PR DESCRIPTION
**Pull request by Ashlita6**
## What is the change?

This PR is to combine the children and parent requirements/implementations and tests into one. 

## Why is the change being made?

Parent requirements are presenting downstream challenges for document review and they aren't needed because they are satisfied by the children. 

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/tests/test_components.py [hunk 1]
```diff
@@ -85,6 +91,12 @@ def test_factory(self):
         self.assertIsInstance(fuelComp.material, materials.UZr)
 
     def test_componentInitializationAndDuplication(self):
+        """Initialize and duplicate a component, veifying the parameters.
```
Comment (keckler) [inline armi/reactor/tests/test_components.py:94]: I don't really understand what this test is trying to prove.

It just creates a bunch of components and then verifies that they can be `deepcopy`-ed? Maybe you can add some clarification to the docstring about what this is supposed to prove?

**Review by keckler — APPROVED** [review]
These impls/tests look good with the corresponding PR on our internal app.

Thanks @Ashlita6 and @opotowsky !

File: armi/reactor/tests/test_components.py (hunk 1 above)
Comment (opotowsky) [reply armi/reactor/tests/test_components.py:94]: You're right....I think the only thing it proves is that a component has parameters...

Comment (john-science) [reply armi/reactor/tests/test_components.py:94]: Correct. The requirement says that a component must have parameters, so this test covers that.  (So do many others, but I picked one.)

The test above proves that components have attributes. That's from the same req.

Do we need to add to the docstring here?

Comment (opotowsky) [reply armi/reactor/tests/test_components.py:94]: I think the docstring repeats what you just said, so it's probably fine?

**Review by john-science — APPROVED** [review]

**Pull request merged**